plugin-process-to-cgroup-bridge/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_keep_config_disable() -> anyhow::Result<()> {
+        let config = Config {
+            processes_metrics: vec!["metric_a".to_string(), "metric_b".to_string()],
+            keep_processed_measurements: false,
+            ..Default::default()
+        };
+
+        let expected = ExpectedCounts {
+            t_initial: 1,
+            t2_initial: 0,
+            t_shared: 2,
+            t_single: 1,
+            t2_shared: 1,
+            t2_single: 0,
+        };
+
+        run_test_with_config(config, expected)
+    }
+
+    #[test]
+    fn test_process_id_not_found_in_procfs() -> anyhow::Result<()> {
+        let base_path = prepare_procfs_mock()?;
+        set_proc_path_for_test(base_path);
+
+        let mut plugins = PluginSet::new();
+
+        let source_config = Config {
+            processes_metrics: vec!["metric_a".to_string(), "metric_b".to_string()],
+            ..Default::default()
+        };
+
+        plugins.add_plugin(PluginInfo {
+            metadata: PluginMetadata::from_static::<DumbNvmlPlugin>(),
+            enabled: true,
+            config: None,
+        });
+        plugins.add_plugin(PluginInfo {
+            metadata: PluginMetadata::from_static::<ProcessToCgroupBridgePlugin>(),
+            enabled: true,
+            config: Some(config_to_toml_table(&source_config)),
+        });
+
+        let make_input = move |ctx: &mut TransformCheckInputContext| -> MeasurementBuffer {
+            let metric = ctx
+                .metrics()
+                .by_name("metric_a")
+                .expect("metric_a metric should exist")
+                .0;
+            let mut m = MeasurementBuffer::new();
+            let point = MeasurementPoint::new_untyped(
+                *T,
+                metric,
+                Resource::LocalMachine,
+                ResourceConsumer::Process { pid: 42 },
+                WrappedMeasurementValue::U64(10),
+            );
+            m.push(point);
+            m
+        };
+
+        let check_output = move |measurements: &MeasurementBuffer| {
+            assert_eq!(measurements.len(), 1);
+            assert_eq!(
+                measurements.iter().next().unwrap().consumer,
+                ResourceConsumer::Process { pid: 42 }
+            );
+        };
+
+        let runtime_expectations = RuntimeExpectations::new().test_transform(
+            TransformName::from_str("process-to-cgroup-bridge", "transform"),
+            make_input,
+            check_output,
+        );
+
+        let agent = agent::Builder::new(plugins)
+            .with_expectations(runtime_expectations)
+            .build_and_start()
+            .unwrap();
+
+        agent.wait_for_shutdown(Duration::from_secs(2)).unwrap();
+
+        Ok(())
+    }
+
+    fn prepare_procfs_mock() -> anyhow::Result<PathBuf> {
+        let tmp = tempdir()?;
+        let base_path = tmp.keep();
```

Comment:
The tempdir should be returned from the function, so that it is cleared after the test. Otherwise it will clutter /tmp.

Reply:
Actually both function have their own responsability.

In the context of extract_process_id_from_measurement, failing/returning error when the measurement has not the right format sounds reasonable.

In the context of cgroup_measurement_from_measurement which call the precedent function, it's also reasonable to "mute" the error because it's expected to have measurements that are not "process" ones and we decide to not treat them (without failing).